soc/arm/nuvoton_npcx/common/soc_i2c.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2020 Nuvoton Technology Corporation.
```

Comment:
This header belongs in `drivers/i2c` and should be `i2c_npcx_controller.h` since it provides the declarations for the functions in that module.  All the functions should be prefixed with `i2c_npcx` as well.

The separation of `i2c_npcx_port` from `i2c_npcx_controller` is also weird. If this is because all the `soc_i2c_` stuff is being imported from an external HAL, why isn't it being kept there, made available to Zephyr through a west module, and referenced like all the other vendors do?

If you do that you can keep the `soc_i2c` prefix, but if it's in Zephyr it really needs to be namespaced by a unique identifier.  IMO.

Reply:
I will move it to "drivers/i2c/" and rename it as "i2c_npcx_controller.h" (like "i2c_ll_stm32.h" did.)
Regarding the separation of i2c_npcx_port from i2c_npcx_controller, it's not related to our HAL. It's based on our i2c hardware architecture. Notice the number of i2c ports and controllers is not equal. (Please refer our i2c port-controller diagram [here](https://github.com/zephyrproject-rtos/zephyr/pull/29144/files#diff-503ccaa7cb43d57ae3e022a9348a1be7c7b78f282a4992da20b6634602fd4ab7R19).) That's why we need the mutex in i2c controller driver in case the other i2c port request influences controller when the state of hardware instance is not idle.

The benefit of this arch is ec can handle i2c master transaction via two ports with one hardware instance. We can offer more i2c ports. Users needn't put too many slave devices on the same bus and improve signal quality by shortening the distance between them.

Reply:
Add one thing.

In npcx 7, there're 10 ports and 8 controllers. Only controller 5 and 6 support two ports, and the others only support a single port. (i2c controller 0/1/2/3/7 supports port 0, and controller 4 supports port 1.) Consider there is no rule between them and the relationship will be changed in the next series, I think separate it into port and controller is easy to maintain.

Reply:
Done for moving and renaming header.